### PR TITLE
roachprod: upgrade go sdk for azure

### PR DESCRIPTION
**Pull request by shailendra-patel**
Upgraded version for azure from v57.1.0 to  v68.0.0
    
Changes as part of this PR:
- Upgraded azure sdk to version v68.0.0.

Epic: none
Informs: https://github.com/cockroachdb/cockroach/issues/129009

Release note: None

### CODE REVIEW

File: pkg/roachprod/vm/azure/azure.go
Comment (renatolabs) [inline pkg/roachprod/vm/azure/azure.go:1704]: Would be nice to include a link to the azure docs where you found out about which machine types support NVMe.

File: pkg/roachprod/vm/azure/utils.go
Comment (renatolabs) [inline pkg/roachprod/vm/azure/utils.go:40]: The `else` block handles a condition that should never happen in practice, right? The "setup local storage" part. Maybe we can remove that logic to make it less confusing.

**Review by renatolabs — COMMENTED** [review]
Looks good, thanks for taking this on!

As a general comment, it would be nice if this work was split in 2 commits: the first one dealing exclusively with the SDK upgrade and no functional changes, while the second one implements NVMe support.

It will be easier to read the diff and will allow us to backport just the upgrade if we want to. Or if something breaks, it gives us an easy way to revert just the NVMe changes while keeping the SDK bump.

**Comment by shailendra-patel** [comment]
@renatolabs I have splitted the work into 2 PR as per your suggestion. Once this upgrade PR is merged, I'll raise separate PR for nvme usage.

**Review by srosenberg — APPROVED** [review]
Let's kick off a smoke test run (SELECT_PROBABILITY=0.3) before the merge.

**Review by renatolabs — APPROVED** [review]
LGTM once we get the results from that run!

**Comment by shailendra-patel** [comment]
I ran the [roachtest azure nightly](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyAzureBazel/16821953?buildTab=overview&hideProblemsFromDependencies=false&expandBuildTestsSection=true&hideTestsFromDependencies=false) for this build. There are three test failures, but I don't believe they are related to the changes in this PR. [@renatolabs](https://github.com/renatolabs) and [@srosenberg](https://github.com/srosenberg), could you please review when you have time? Let me know if I should proceed with the merge.

**Comment by renatolabs** [comment]
For the record: a few clusters could not be created on the run above with an error like the following:

> ```Unable to create (or reuse) cluster for test sql-stats/mixed-version due to: in provider: azure: creating VM teamcity-16821953-1725979909-59-n5cpu4-0005: Code="AllocationFailed" Message="The VM allocation failed due to an internal error. Please retry later or try deploying to a different location.".```

Unclear what this means exactly, but seems related to the SDK upgrade, as I can't find similar errors on past `master` runs.

We should keep an eye out for this over the next couple of days to see how often it happens.

**Comment by shailendra-patel** [comment]
For below two tests cluster could not be created with an error message mentioned in above comment:
1. unoptimized-query-oracle/disable-rules=all/rand-multi-region
2. sql-stats/mixed-version
@renatolabs I'll try to run roachtest locally with debugger mode to figure out is this because of upgrade. Thanks for pointing out.

**Comment by shailendra-patel** [comment]
From azure docs seems resource crunch.

> Error message: "Allocation failed. We do not have sufficient capacity for the requested VM size in this region. Read more about improving likelihood of allocation success at https://aka.ms/allocation-guidance"
Azure doc l[ink](https://learn.microsoft.com/en-us/troubleshoot/azure/virtual-machines/windows/allocation-failure)

**Comment by renatolabs** [comment]
Yeah, I saw that link, but there are two kinds of `Allocation failed` errors [1]: one due to insufficient capacity (as you mentioned), and the internal error one we observed. I have a feeling they are different (`internal error` would be a weird way to report insufficient capacity, after all).

[1] https://learn.microsoft.com/en-us/troubleshoot/azure/virtual-machines/windows/error-messages#common-virtual-machine-management-errors

**Comment by shailendra-patel** [comment]
In latest run of [azure nightly ](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyAzureBazel/16846942?buildTab=log&focusLine=8020&logView=flowAware&linesState=2566.7758.7800.8856.8971.8987.9057.9302.9567)at `12 Sep 05:02` i do not see errors related to `AllocationFailed`. I see below error message in creating cluster which has occurred in previous run also before sdk upgrade.
```
Unable to create (or reuse) cluster for test costfuzz/rand-multi-region due to: one or more parallel execution failure(s): COMMAND_PROBLEM: exit status 1.
```

**Pull request merged**
